### PR TITLE
MODORDERS-783 implemented export history proxy API

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
* https://issues.folio.org/browse/MODORDERS-783

## Approach
* implemented export history proxy API for search queries

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/config/ApplicationConfig.java [hunk 1]
```diff
@@ -36,48 +29,15 @@
...
-import org.folio.service.finance.transaction.EncumbranceWorkflowStrategyFactory;
-import org.folio.service.finance.transaction.EncumbrancesProcessingHolderBuilder;
-import org.folio.service.finance.transaction.OpenToClosedEncumbranceStrategy;
-import org.folio.service.finance.transaction.OpenToPendingEncumbranceStrategy;
-import org.folio.service.finance.transaction.PendingToOpenEncumbranceStrategy;
-import org.folio.service.finance.transaction.PendingToPendingEncumbranceStrategy;
-import org.folio.service.finance.transaction.ReceivingEncumbranceStrategy;
-import org.folio.service.finance.transaction.TransactionService;
-import org.folio.service.finance.transaction.TransactionSummariesService;
+import org.folio.service.finance.transaction.*;
```
Comment (Abdulkhakimov) [inline src/main/java/org/folio/config/ApplicationConfig.java:32]: Could you replace wildcard import to common?

File: src/test/java/org/folio/ApiTestSuite.java [hunk 2]
```diff
@@ -23,16 +23,7 @@
...
-import org.folio.rest.impl.CheckinReceivingApiTest;
-import org.folio.rest.impl.HoldingsSummaryAPITest;
-import org.folio.rest.impl.OrderTemplateTest;
-import org.folio.rest.impl.PieceApiTest;
-import org.folio.rest.impl.PoNumberApiTest;
-import org.folio.rest.impl.PurchaseOrderLinesApiTest;
-import org.folio.rest.impl.PurchaseOrdersApiTest;
-import org.folio.rest.impl.ReceivingHistoryApiTest;
-import org.folio.rest.impl.TitlesApiTest;
+import org.folio.rest.impl.*;
```
Comment (Abdulkhakimov) [inline src/test/java/org/folio/ApiTestSuite.java:26]: the same question with wildcard import

**Review by Abdulkhakimov — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -1523,6 +1539,19 @@
...
     },
+    {
+      "permissionName" : "orders.export-history.collection.get",
+      "displayName" : "orders-export-history-collection get",
+      "description" : "Fetch export history collection"
+    },
+    {
+      "permissionName" : "orders.export-history.all",
+      "displayName" : "All orders suffixes perms",
+      "description" : "All permissions for the orders configuration - suffixes",
```
Comment (SerhiiNosko) [inline descriptors/ModuleDescriptor-template.json:1550]: Please change dispaly name and description

File: pom.xml [hunk 4]
```diff
@@ -278,6 +278,12 @@
       <version>2.27.2</version>
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>org.projectlombok</groupId>
+          <artifactId>lombok</artifactId>
+          <version>1.18.24</version>
+          <scope>compile</scope>
+      </dependency>
```
Comment (SerhiiNosko) [inline pom.xml:286]: seems spaces should be corrected

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**